app/controllers/ontologies_controller.rb

```diff
@@ -437,6 +437,67 @@ def metrics_evolution
     render partial: 'ontologies/sections/metadata/metrics_evolution_graph', locals: { data: data }
   end
 
+  def selector
+    @categories = LinkedData::Client::Models::Category.all(display_links: false, display_context: false)
+    @groups = LinkedData::Client::Models::Group.all(display_links: false, display_context: false)
+    @filters = ontology_filters_init(@categories, @groups)
+    @select_id = params[:id]
+    render 'ontologies/selector/selector' , layout: false
+  end
+
+  def selector_results
+    api_params = {}
+    if params[:showOntologyViews]
+      api_params[:include_views] = 'true'
+    end
+    @ontologies = LinkedData::Client::HTTP.get('/ontologies', api_params)
+    @total_ontologies_number = @ontologies.length
+    @input = params[:input] || ''
+    @ontologies = @ontologies.select { |ontology| ontology.name.downcase.include?(@input.downcase) || ontology.acronym.downcase.include?(@input.downcase)}
+
+    if params[:groups] 
+      @ontologies = @ontologies.select do |ontology|
+        (ontology.group & params[:groups]).any?
+      end
+    end
+
+    if params[:categories]
+      @ontologies = @ontologies.select do |ontology|
+        (ontology.hasDomain & params[:categories]).any?
+      end
+    end
+
+    if params[:formats] || params[:naturalLanguage] || params[:formalityLevel] || params[:isOfType] || params[:showRetiredOntologies]
+      api_params = {'display': 'all'}
+      if params[:showOntologyViews]
+        api_params[:include_views] = 'true'
+      end
+      submissions = LinkedData::Client::HTTP.get('/submissions', api_params)
```

Comment:
don't use ` LinkedData::Client::HTTP.get` for already defined models, so here it will be  `LinkedData::Client::OntologySubmission.all(api_params)`